tests/test_krylov.py

```diff
@@ -0,0 +1,304 @@
+@pytest.mark.parametrize(
+    "input_kry_gensub",
+    [
+        (
+            10.2,
+            100,
+            True,
+            True,
+            "The maximum size of Krylov subspace is required to be greater than 0",
+        ),
+        (
+            -2,
+            100,
+            True,
+            True,
+            "The maximum size of Krylov subspace is required to be greater than 0",
+        ),
+        (
+            10,
+            100.3,
+            True,
+            True,
+            "The frequency value for building the Krylov subspace is required to be greater or equal to 0 Hz",
+        ),
+        (
+            10,
+            -100,
+            True,
+            True,
+            "The frequency value for building the Krylov subspace is required to be greater or equal to 0 Hz",
+        ),
+        (
+            10,
+            100,
+            3,
+            False,
+            "The chk_ortho_key value for building the Krylov subspace is required to be Boolean True or False",
+        ),
+        (
+            10,
+            100,
+            True,
+            2,
+            "The out_key value for building the Krylov subspace is required to be to be Boolean True or False",
+        ),
+    ],
+)
+def test_non_valid_inputs_krygensub(mapdl, input_kry_gensub):
+    if not meets_version(mapdl._server_version, (0, 5, 0)):
+        pytest.skip("Requires MAPDL v222 and above")
+
+    mapdl.clear()
+    # Parameters set for Krylov
+    maxQ = input_kry_gensub[0]
+    freq_val = input_kry_gensub[1]
+
+    mapdl.cdread("db", os.path.join(lib_path, "krylov_pressure_load"), "cdb")
+    solu_krylov(mapdl, 100)
+    dd = mapdl.krylov
+    with pytest.raises(ValueError) as e:
+        dd.krygensub(maxQ, freq_val, input_kry_gensub[2], input_kry_gensub[3])
+    assert str(e.value) == input_kry_gensub[4]
+
+
+@pytest.mark.parametrize(
+    "input_krysolve",
+    [
+        (
+            -2,
+            100,
+            10,
+            0,
+            True,
+            "The beginning frequency value for solving the reduced solution is required to be greater than or equal to 0",
+        ),
+        (
+            2,
+            100,
+            -10,
+            1,
+            False,
+            "The number of frequencies for which to compute the reduced solution is required to be greater than 0",
+        ),
+        (
+            2,
+            100,
+            10,
+            -1,
+            True,
+            "The Load key value for computing the reduced solution is required to be 0 or 1",
+        ),
+        (
+            2,
+            100,
+            10,
+            0,
+            2,
+            "The out_key value for computing the reduced solution is required to be Boolean True or False",
+        ),
+    ],
+)
+def test_non_valid_inputs_krysolve(mapdl, input_krysolve):
+    if not meets_version(mapdl._server_version, (0, 5, 0)):
+        pytest.skip("Requires MAPDL v222 and above")
```

Comment:
```suggestion
        pytest.skip("Requires MAPDL 2022 R2 or later.")
```

Reply:
Public exposure of version is supposed to be formatted per my edit.

Reply:
local.